fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/properties/RelationalExpressionDepthProperty.java

```diff
@@ -90,4 +99,11 @@ public int evaluate(@Nonnull RelationalExpression expression) {
         return expression.acceptPropertyVisitor(this);
     }
 
+    @Nonnull
+    @Override
+    @SpotBugsSuppressWarnings("NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE")
+    public Integer evaluateAtQuantifier(@Nonnull final Quantifier quantifier, @Nullable final Integer rangesOverResult) {
+        // since we do visit expression references in this property we can insist on rangesOverResult not being null
```

Comment:
Given how many of these there are, perhaps we should adjust the `PlannerProperty` interface to have a version of this method where `rangesOverResult` is necessarily non-null? I'm happy to do that in a follow-up PR.

Reply:
So this one is interesting, I had it like that, the problem is the following:

`evaluateAtQuantifier` can be called on `null` but must return non null. This cannot be done in the generic sense other than bombing by means of e.g. `requireNonNull()`. Yes, you could put that into the default implementation, however, that would cause this method to immediately throw an exception if `shouldVisit` is `false` for expression refs. That seems to behave weird. So I left it up to the implementor to deal with this. As all of them return for that `shouldVisit(expressionRef)` `false` they can in fact assume there is an internal problem if the method gets called on null.

What we could do instead is to default all `shouldVisit()` in the interface and set them to `true`. We then could point out that the default implementation would always visit everything and that if you want to override, then you must deal with the fallout of having an `evaluate...` to be called with `null`s.